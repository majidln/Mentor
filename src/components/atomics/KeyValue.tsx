import React, {FC, ReactElement} from 'react';
import {View, Text, StyleSheet} from 'react-native';

export interface KeyValueProps {
  label?: string;
  value?: string;
}

export const KeyValue: FC<KeyValueProps> = ({
  label = '',
  value = '',
}): ReactElement => {
  return (
    <View style={styles.wrapper}>
      <Text style={styles.label}>{label}:</Text>
      <Text style={styles.value}>{value}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
  },
  label: {
    fontSize: 16,
    fontWeight: '400',
    paddingRight: 4
  },
  value: {
    fontSize: 16,
    fontWeight: '700',
  },
});
