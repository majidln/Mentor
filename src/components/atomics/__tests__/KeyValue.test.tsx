import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';

import {KeyValue} from './../KeyValue';

describe('Test for atomic KeyValue', () => {
  const label = 'name: ';
  const value = 'Majid';

  it('KeyValue render correctly two text', () => {
    const testRenderer = renderer.create(
      <KeyValue label={label} value={value} />,
    );
    const testInstance = testRenderer.root;

    const textElements = testInstance.findAllByType(Text);

    expect(textElements.length).toBe(2);
  });

  it('KeyValue Snapshot renders correctly', () => {
    const tree = renderer
      .create(<KeyValue label={label} value={value} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
