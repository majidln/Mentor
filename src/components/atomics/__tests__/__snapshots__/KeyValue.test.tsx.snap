// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for atomic KeyValue KeyValue Snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "400",
        "paddingRight": 4,
      }
    }
  >
    name: 
    :
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Majid
  </Text>
</View>
`;
